streams/src/main/java/org/apache/kafka/streams/kstream/Materialized.java

```diff
@@ -34,9 +34,23 @@
 
 /**
  * Used to describe how a {@link StateStore} should be materialized.
- * You can either provide a custom {@link StateStore} backend
- * through one of the provided methods accepting a supplier or use the default RocksDB backends
- * by providing just a store name.
+ * You can either provide a custom {@link StateStore} backend through one of the provided methods accepting a supplier
+ * or use the default RocksDB backends by providing just a store name.
+ * <p>
+ * For example, you can read a topic as {@link KTable} and force a state store materialization to access the content
+ * via Interactive Queries API:
+ * <pre>{@code
+ * StreamsBuilder builder = new StreamsBuilder();
+ * KTable<Integer, Integer> table = builder.table(
+ *   "topicName",
+ *   Materialized.as("queryable-store-name"));
+ * }</pre>
+ *
+ * @param <K> type of record key
+ * @param <V> type of record value
+ * @param <S> type of state store (note, state stores always have key/value types {@code <Bytes,byte[]>}
```

Comment:
Style nit for "note" and whitespace between `Bytes` and `byte[]`:

> (note, state stores always have key/value types {@code <Bytes,byte[]>}

should be

> (note: state stores always have key/value types {@code <Bytes, byte[]>}

Reply:
Question: If the state stores are always of type `<Bytes, byte[]>`, why does the user even have to specify the type explicitly?

Reply:
The key and value types `K` and `V` are not `Bytes`/`byte[]` but can be anything -- only the store `S` has a "nested" generic type that is `<Bytes,byte[]>`

I don't like the WS in `<Bytes,byte[]>` and keep as is.